src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -0,0 +1,290 @@
+                {
+                    if (IsValidIssuer(validIssuerTemplate, tenantId, issuer))
+                    {
+                        return issuer;
+                    }
+                }
+            }
+
+            if (validationParameters.ValidIssuer != null)
+            {
+                if (IsValidIssuer(validationParameters.ValidIssuer, tenantId, issuer))
+                {
+                    return issuer;
+                }
+            }
+
+            try
+            {
+                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
+                {
+                    if (AadIssuerV2 == null)
+                    {
+                        IssuerMetadata issuerMetadata =
+                            CreateConfigManager(AadAuthority).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+                        AadIssuerV2 = issuerMetadata.Issuer;
+                    }
+
+                    if (IsValidIssuer(AadIssuerV2, tenantId, issuer))
+                    {
+                        return issuer;
+                    }
+                }
+                else
+                {
+                    if (AadIssuerV1 == null)
+                    {
+                        IssuerMetadata issuerMetadata =
+                            CreateConfigManager(CreateV1Authority()).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+                        AadIssuerV1 = issuerMetadata.Issuer;
+                    }
+
+                    if (IsValidIssuer(AadIssuerV1, tenantId, issuer))
+                    {
+                        return issuer;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidIssuerException(
+                string.Format(
+                    CultureInfo.InvariantCulture,
+                    LogMessages.IDX40103,
+                    issuer),
+                ex));
+            }
+
+            // If a valid issuer is not found, throw
+            throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidIssuerException(
+                string.Format(
+                    CultureInfo.InvariantCulture,
+                    LogMessages.IDX40103,
+                    issuer)));
+        }
+
+        private string CreateV1Authority()
+        {
+            if (AadAuthority.Contains(AadIssuerValidatorConstants.Organizations))
+            {
+                return AadAuthority.Replace($"{AadIssuerValidatorConstants.Organizations}/v2.0", AadIssuerValidatorConstants.Common);
+            }
+
+            return AadAuthority.Replace("/v2.0", string.Empty);
+        }
+
+        private ConfigurationManager<IssuerMetadata> CreateConfigManager(
+            string aadAuthority)
+        {
+            if (HttpClient != null)
```

Comment:
We should be using the HttpClient from BaseConfigurationManager, we will need to expose that.
The asp.net team asked us to make sure all calls use that HttpClient.

Reply:
that's cool. we can have a static method then.